### PR TITLE
api updates

### PR DESCRIPTION
**Pull request by fabrixxm**

### CODE REVIEW

File: include/api.php [hunk 1]
```diff
@@ -664,18 +664,20 @@ function api_item_get_user(&$a, $item) {
 
 		// Comments in threads may appear as wall-to-wall postings.
 		// So only take the owner at the top posting.
-		if ($item["id"] == $item["parent"])
-			$status_user = api_get_user($a,$item["owner-link"]);
-		else
-			$status_user = api_get_user($a,$item["author-link"]);
-
+		#if ($item["id"] == $item["parent"])
```
Comment (tobiasd) [inline include/api.php]: I did not know that was possible in PHP :-D you made my day Fabio!

File: include/api.php [hunk 2]
```diff
@@ -664,18 +664,20 @@ function api_item_get_user(&$a, $item) {
...
 		$status_user["protected"] = (($item["allow_cid"] != "") OR
 						($item["allow_gid"] != "") OR
 						($item["deny_cid"] != "") OR
 						($item["deny_gid"] != "") OR
 						$item["private"]);
 
-		return ($status_user);
+		$owner_user = api_get_user($a,$item["owner-link"]);						
+						
+		return (array($status_user, $owner_user));
```
Comment (tobiasd) [inline include/api.php]: I think the docs should be extended now, when the function returns more data then before.

File: include/api.php (hunk 1 above)
Comment (annando) [inline include/api.php]: I don't really like this kind of commenting out stuff. What is the benefit of not using "//"?

**Comment by annando** [comment]
Have you checked against Twidere and AndStatus if both are working and if reshared items are displayed correctly?

File: include/api.php (hunk 1 above)
Comment (fabrixxm) [inline include/api.php]: > What is the benefit of not using "//"?
nothing, just habit. 
I'll update this PR and I'll remove the commented lines

**Comment by fabrixxm** [comment]
> Have you checked against Twidere and AndStatus if both are working and if reshared items are displayed correctly?

No, but looks like more inline with [twitter faq](https://dev.twitter.com/faq#76)

> [...]The only fields typically used in the outermost object are the ID, user information about the retweeter, and the created_at date for the retweet event.[...]

So the body and the user info of the outermost item remains unchanged. And clients that don't use 'retweeted_status' will see the post as  '[recycle icon] [original text]' from [resharer user] , which is better than the current status were we see only '[original text]' from [original user], which is confusing...

I'll try to install Twidere and AndStatus and check if it works correctly...

**Comment by tobiasd** [comment]
It should be possible to install them with alien, but friendly is nicer and more intuitive :-)

**Comment by fabrixxm** [comment]
This is Twidere showing a reshare ([this one](https://kirgroup.com/display/96008cc2705787a16399e99952130260)) with this version of `api.php`

screenshot-16-07-14-17-00-12

The post is shown like the original, with a little notice on top "ReTwitted by Sandro".
AndStatus work the same.

**Comment by annando** [comment]
My plan is to completely rework the XML generation of the API functions. I think we should generate the JSON at first and then should use a function to create the XML from the JSON data. (Hopefully the structure is identical, is it?)

We can drop the support for activity streams since there only was a single (long deprecated) application that used it.

Maybe I will find some time in the next days to do this work.

File: include/api.php [hunk 3]
```diff
@@ -2407,15 +2442,31 @@ function api_format_items($r,$user_info, $filter_user = false) {
...
+
+			if ($item["id"] == $item["parent"]) {
+				$retweeted_item = api_share_as_retweet($item);
+				if ($retweeted_item !== false) {
+					$retweeted_status = $status;
+					try {
+						$retweeted_status["user"] = api_get_user($a,$retweeted_item["author-link"]);
+					} catch( BadRequestException $e ) {
+						// user not found. should be found?
+						// TODO: check if the user should be found...
```
Comment (rabuzarus) [inline include/api.php]: please use `@todo` instead. So it can be recognized by doxygen

File: include/api.php [hunk 4]
```diff
@@ -2974,23 +3025,36 @@ function api_friendica_remoteauth(&$a) {
...
 
 		$attributes = preg_replace("/\[share(.*?)\]\s?(.*?)\s?\[\/share\]\s?/ism","$1",$body);
-		// Skip if there is no shared message in there
-		if ($body == $attributes)
-			return(false);
+ 		// Skip if there is no shared message in there
+ 		if ($body == $attributes)
+			return false;
+
+		// NOTE: we could check te guid="" attribute and then  try to load original post
```
Comment (annando) [inline include/api.php]: There is this function: https://github.com/friendica/friendica/blob/develop/include/diaspora.php#L2739

it does a check for a repeated item and should be ideal for this. I really should have replaced this function a long time ago ...

See how it is implement here: https://github.com/friendica/friendica/blob/develop/include/diaspora.php#L2027-L2029

Only problem: We maybe should move this function at another place.

**Comment by annando** [comment]
Like previously announced, today I wrote a prototype for the automatic XML generation without a template. There is still a lot of work left. When your pull request is done I will have a look at the whole stuff and will see if my changes will work reliably.

**Comment by annando** [comment]
I just realized that we don't seem to have a general function for fetching the "share" data. Shame on me. (I really thought so, but the diaspora function does only a subset and the function in bbcode.php can't be used)

@rabuzarus @tobiasd Are you satisfied with the pull request? Then we should accept it by now.

**Comment by rabuzarus** [comment]
I'm OK with it

**Comment by tobiasd** [comment]
I miss the `#` ;-)

**Pull request merged**